From 3497534445e20b0dd6ca2f3fe51b308d1e30e6ed Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Fri, 13 Apr 2018 13:25:04 +0300
Subject: [PATCH 259/741] bus: fsl-mc: move fsl_mc_device_remove to common
 header

commit 60f8ebf17ef336ba8332f9215cd279f22b82e806 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

VFIO fsl-mc driver need this function on device remove,
moving this to common header file

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@nxp.com>
Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/bus/fsl-mc/fsl-mc-private.h | 2 --
 include/linux/fsl/mc.h              | 2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/bus/fsl-mc/fsl-mc-private.h b/drivers/bus/fsl-mc/fsl-mc-private.h
index 326671eb520e..e1b43629c140 100644
--- a/drivers/bus/fsl-mc/fsl-mc-private.h
+++ b/drivers/bus/fsl-mc/fsl-mc-private.h
@@ -548,8 +548,6 @@ int __must_check fsl_mc_device_add(struct fsl_mc_obj_desc *obj_desc,
 				   const char *driver_override,
 				   struct fsl_mc_device **new_mc_dev);
 
-void fsl_mc_device_remove(struct fsl_mc_device *mc_dev);
-
 int __init dprc_driver_init(void);
 
 void dprc_driver_exit(void);
diff --git a/include/linux/fsl/mc.h b/include/linux/fsl/mc.h
index 01f10735dfbb..06887d3ccb61 100644
--- a/include/linux/fsl/mc.h
+++ b/include/linux/fsl/mc.h
@@ -366,6 +366,8 @@ int mc_send_command(struct fsl_mc_io *mc_io, struct fsl_mc_command *cmd);
 	module_driver(__fsl_mc_driver, fsl_mc_driver_register, \
 		      fsl_mc_driver_unregister)
 
+void fsl_mc_device_remove(struct fsl_mc_device *mc_dev);
+
 /*
  * Macro to avoid include chaining to get THIS_MODULE
  */
-- 
2.17.1

